espressif/vscode-esp-idf-extension
# Add check for encryption in sdkconfig

## Description

Added check for SDK config modifications for encryption.

Fixes #1067 

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

## Steps to test this pull request

1. Open a project
2. Use "ESP-IDF: SDK Configuration editor (menuconfig)
3. Enable "Enable flash encryption on boot (READ DOCS FIRST)"
4. Build and Flash project with UART
5. Flash should be successful and the task should contain the --encrypt flag
EX:

> Executing task: C:\xxxxx\.espressif\python_env\idf5.0_py3.8_env\Scripts\python.exe c:\xxxxxx\esp-idf-v5.0.1\components\esptool_py\esptool\esptool.py -p COM5 -b 460800 --before default_reset --after no_reset --chip esp32s3 write_flash --flash_mode dio --flash_freq 80m **--encrypt** --flash_size 16MB 0x0 bootloader/bootloader.bin 0x20000 app.bin 0x8000 partition_table/partition-table.bin

**Test Configuration**:
* ESP-IDF Version: 5.1
* OS (Windows,Linux and macOS): Mac

## Checklist
